test/lib/pure-helpers.test.js

```diff
@@ -0,0 +1,215 @@
+  expect(typeof helpers.servicePropertiesToServiceSubscriptionPayload).toBe('function')
+  expect(typeof helpers.getCertFilesLocation).toBe('function')
+  expect(typeof helpers.findOrgOrThrow).toBe('function')
+  expect(typeof helpers.findProjectOrThrow).toBe('function')
+  expect(typeof helpers.findWorkspaceOrThrow).toBe('function')
+  expect(typeof helpers.getAddServicesOperationPromptChoices).toBe('function')
+  expect(typeof helpers.enhanceWorkspaceConfiguration).toBe('function')
+  expect(typeof helpers.workspaceNamesToPromptString).toBe('function')
+})
+
+describe('orgsToPromptChoices', () => {
+  test('with input that has entp and non entp orgs', () => {
+    expect(helpers.orgsToPromptChoices(dataMocks.organizations))
+      .toEqual(dataMocks.promptChoices.orgs)
+  })
+})
+
+describe('projectsToPromptChoices', () => {
+  test('with input that has deleted, not enabled and non-jaeger projects', () => {
+    // expects in reverse order (latest created first)
+    expect(helpers.projectsToPromptChoices(dataMocks.projects))
+      .toEqual(dataMocks.promptChoices.projects)
+  })
+})
+
+describe('workspacesToPromptChoices', () => {
+  test('with some workspaces that are not enabled and w/o runtime namespace workspaces', () => {
+    // expects in reverse order (latest created first)
+    expect(helpers.workspacesToPromptChoices(dataMocks.workspaces))
+      .toEqual(dataMocks.promptChoices.workspaces)
+  })
+})
+
+describe('licenseConfigsToPromptChoices', () => {
+  test('with input licenseConfigs', () => {
+    const licenseConfigs = dataMocks.serviceProperties[1].licenseConfigs
+    expect(helpers.licenseConfigsToPromptChoices(licenseConfigs))
+      .toEqual(dataMocks.promptChoices.licenseConfigs[1])
+  })
+})
+
+describe('servicesToPromptChoices', () => {
+  test('with some not enabled services and some not entp services', () => {
+    expect(helpers.servicesToPromptChoices(dataMocks.services))
+      .toEqual(dataMocks.promptChoices.services)
+  })
+})
+
+test('filterEnabledServices', () => {
+  expect(helpers.filterEnabledServices(dataMocks.services)).toEqual(dataMocks.enabledServices)
+})
+
+describe('findFirstEntpCredential', () => {
+  test('when there is one', () => {
+    expect(helpers.findFirstEntpCredential(dataMocks.integrations))
+      .toEqual(dataMocks.integrations[2])
+  })
+  test('when there is none', () => {
+    expect(helpers.findFirstEntpCredential([])).toEqual(undefined)
+  })
+})
+
+test('servicePropertiesToNames', () => {
+  const serviceProperties = dataMocks.serviceProperties
+  expect(helpers.servicePropertiesToNames(serviceProperties)).toEqual([
+    serviceProperties[0].name, serviceProperties[1].name, serviceProperties[2].name
+  ])
+})
+
+describe('fixServiceProperties', () => {
+  test('with serviceProperties that need fixing', () => {
+    const allLicenseConfigs = dataMocks.serviceProperties
+      .filter(sp => !!sp.licenseConfigs)
+      .flatMap(sp => sp.licenseConfigs)
```

Comment:
I just added the GH Actions to run the npm test for the build matrix for nodes 10, 12, 14.
It is failing on `flatMap` which is not in node 10.

Reply:
thanks for catching this @shazron, will fix